backend/tests/Tests/SqlCompiler.Tests.fs

```diff
@@ -184,6 +184,155 @@ let inlineWorksWithNested =
     return! result |> Ply.toTask
   }
 
+let inlineWorksWithPackageFunctionsSqlBinOp =
+  testTask "inlineWorksWithPackageFunctionsSqlBinOp" {
+    let! state =
+      executionStateFor
+        (System.Guid.NewGuid())
+        false
+        false
+        Map.empty
+        Map.empty
+        Map.empty
+        Map.empty
+    let fns = ExecutionState.availableFunctions state
+    let! expr =
+      p
+        "let x = true in (let y = false in (PACKAGE.Darklang.Stdlib.Bool.and_v0 x y))"
+
+    let! expected = p "true && false"
+    let result =
+      uply {
+        let! result = C.inline' fns "value" Map.empty expr
+        return Expect.equalExprIgnoringIDs result expected
+      }
+    return! result |> Ply.toTask
+  }
+
+let inlineWorksWithPackageFunctionsSqlFunction =
+  testTask "inlineWorksWithPackageFunctionsSqlFunction" {
+    let! state =
+      executionStateFor
+        (System.Guid.NewGuid())
+        false
+        false
+        Map.empty
+        Map.empty
+        Map.empty
+        Map.empty
+    let fns = ExecutionState.availableFunctions state
+    let! expr =
+      p
+        """let x = "package" in (let y = "e" in (PACKAGE.Darklang.Stdlib.String.replaceAll_v0 x y "es"))"""
+
+    let! expected = p """Builtin.String.replaceAll_v0 "package" "e" "es" """
+    let result =
+      uply {
+        let! result = C.inline' fns "value" Map.empty expr
+        return Expect.equalExprIgnoringIDs result expected
+      }
+    return! result |> Ply.toTask
+  }
+
+
+let inlineWorksWithUserFunctions =
+  testTask "inlineWorksWithUserFunctions" {
+    let! state =
+      executionStateFor
+        (System.Guid.NewGuid())
+        false
+        false
+        Map.empty
+        Map.empty
+        Map.empty
+        Map.empty
+    let fns = ExecutionState.availableFunctions state
+    let! expr =
+      p
+        "let a = 1 in let b = 9 in let c = userAdd 6 4 in (p.height == c) && (p.age == b)"
+
+    let! expected = p "p.height == userAdd 6 4 && p.age == 9"
```

Comment:
I don't think we can define a function (`let userAdd = ...`) in `let! expr` . So, I am not sure where to put the definition of `userAdd` to be able to inline it.